stdpopsim/slim_engine.py

```diff
@@ -1363,24 +1363,28 @@ def matrix2str(
     sampling_episodes = []
     for sample_set in samples:
         pop = demographic_model.model[sample_set.population]
-        # We're currently sampling genomes, and set the ploidy to 1 to make
-        # sure it all fits together. This is probably confusing though and
-        # maybe we should change it.
-        assert sample_set.ploidy == 1
-        # SLiM can only sample individuals, which we assume are diploid.
         count = sample_set.num_samples
         time = 0 if sample_set.time is None else sample_set.time
         time = round(time * demographic_model.generation_time)
-        n_inds = (count + 1) // 2
-        if count % 2 != 0:
-            gen = time / demographic_model.generation_time
-            warnings.warn(
-                stdpopsim.SLiMOddSampleWarning(
-                    f"SLiM simulates diploid individuals, so {n_inds} "
-                    f"individuals will be sampled for the {count} haploids "
-                    f"requested from population {pop.name} at time {gen}. "
-                    "See #464."
+        if sample_set.ploidy == 1:
+            # SLiM can only sample individuals, which we assume are diploid.
+            n_inds = (count + 1) // 2
```

Comment:
I guess this results in a tree sequence where pairs of haploid individuals are grouped to look like single individuals? Can this be fixed by post-processing?

Reply:
Maybe just open an issue for this instead of tackling it here.

Reply:
Good point! Mentioned in #1371.